server/src/internalClusterTest/java/org/elasticsearch/search/SearchCancellationIT.java

```diff
@@ -221,43 +224,12 @@ public void testCancelMultiSearch() throws Exception {
         }
     }
 
-    /**
-     * The test `testCancelFailedSearchWhenPartialResultDisallowed` usually fails when concurrency is enabled unless
-     * the `cancelledLatch.await()` section is commented out. However, this approach seems prone to race conditions.
-     * Further investigation is needed to determine if this test just needs to be revised, or rather, if it is
-     * detecting a deeper issue.  For now, we will disable concurrency here.
-     */
-    @Override
-    protected boolean enableConcurrentSearch() {
-        return false;
-    }
-
     public void testCancelFailedSearchWhenPartialResultDisallowed() throws Exception {
-        final List<ScriptedBlockPlugin> plugins = initBlockFactory();
         int numberOfShards = between(2, 5);
-        AtomicBoolean failed = new AtomicBoolean();
-        CountDownLatch queryLatch = new CountDownLatch(1);
-        CountDownLatch cancelledLatch = new CountDownLatch(1);
-        for (ScriptedBlockPlugin plugin : plugins) {
-            plugin.disableBlock();
-            plugin.setBeforeExecution(() -> {
-                try {
-                    queryLatch.await(); // block the query until we get a search task
-                } catch (InterruptedException e) {
-                    throw new AssertionError(e);
-                }
-                if (failed.compareAndSet(false, true)) {
-                    throw new IllegalStateException("simulated");
-                }
-                try {
-                    cancelledLatch.await(); // block the query until the search is cancelled
-                } catch (InterruptedException e) {
-                    throw new AssertionError(e);
-                }
-            });
-        }
         createIndex("test", numberOfShards, 0);
         indexTestData();
+
+        // Define (but don't run) the search request, expecting a partial shard failure. We will run it later.
```

Comment:
nit: You can move this definition down to where we execute search to remove this comment.

Reply:
I see your point, though I think it improves the overall readability to place this request definition up front (in logical order), as it gives the context needed to understand the purpose of the following Blocking Scripts.